HiAnalysis/HiOnia/plugins/HiOniaAnalyzer.cc

```diff
@@ -799,451 +864,696 @@ HiOniaAnalyzer::fillTreeMuon(const pat::Muon* muon, int iType, ULong64_t trigBit
+      }
+      else {
+	cout << "HiOniaAnalyzer::fillTreeJpsi: no PVfor muon pair stored" << endl;
+	return;
+      }
 
-    Reco_QQ_mupl_isGoodMuon[Reco_QQ_size] = muon::isGoodMuon(*muon1, muon::TMOneStationTight);
-    Reco_QQ_mumi_isGoodMuon[Reco_QQ_size] = muon::isGoodMuon(*muon2, muon::TMOneStationTight);
+      new((*Reco_QQ_vtx)[Reco_QQ_size])TVector3(RefVtx.X(),RefVtx.Y(),RefVtx.Z());
 
-    Reco_QQ_mupl_TrkMuArb[Reco_QQ_size] = muon1->muonID("TrackerMuonArbitrated");
-    Reco_QQ_mupl_TMOneStaTight[Reco_QQ_size] = muon1->muonID("TMOneStationTight");
+      TLorentzVector vMuon1 = lorentzMomentum(muon1->p4());
+      TLorentzVector vMuon2 = lorentzMomentum(muon2->p4());
 
-    Reco_QQ_mumi_TrkMuArb[Reco_QQ_size] = muon2->muonID("TrackerMuonArbitrated");
-    Reco_QQ_mumi_TMOneStaTight[Reco_QQ_size] = muon2->muonID("TMOneStationTight");
+      reco::TrackRef iTrack_mupl;
+      reco::TrackRef iTrack_mumi;
 
-    iTrack_mupl = muon1->innerTrack();
-    iTrack_mumi = muon2->innerTrack();
+      if (muon1->charge() > muon2->charge()) {
 
-    if (muon1->isGlobalMuon())
-      gTrack_mupl = muon1->globalTrack();
+	Reco_QQ_mupl_idx[Reco_QQ_size] = IndexOfThisMuon(&vMuon1 , Reco_mu_4mom, false); 
+	Reco_QQ_mumi_idx[Reco_QQ_size] = IndexOfThisMuon(&vMuon2 , Reco_mu_4mom, false);
 
-    if (muon2->isGlobalMuon())
-      gTrack_mumi = muon2->globalTrack();
-  }
-  else {
-    new((*Reco_QQ_mupl_4mom)[Reco_QQ_size])TLorentzVector(vMuon2);
-    new((*Reco_QQ_mumi_4mom)[Reco_QQ_size])TLorentzVector(vMuon1);
+	if (TVector2::Phi_mpi_pi(vMuon1.Phi() - vMuon2.Phi()) > 0) Reco_QQ_isCowboy[Reco_QQ_size] = true;
+	else Reco_QQ_isCowboy[Reco_QQ_size] = false;
 
-    Reco_QQ_mupl_trig[Reco_QQ_size] = trigBits_mu2;
-    Reco_QQ_mumi_trig[Reco_QQ_size] = trigBits_mu1;
+	iTrack_mupl = muon1->innerTrack();
+	iTrack_mumi = muon2->innerTrack();
 
-    Reco_QQ_mupl_SelectionType[Reco_QQ_size] = muonIDmask(muon2);
-    Reco_QQ_mumi_SelectionType[Reco_QQ_size] = muonIDmask(muon1);
+      }
+      else {
 
-    if (TVector2::Phi_mpi_pi(vMuon2.Phi() - vMuon1.Phi()) > 0) Reco_QQ_isCowboy[Reco_QQ_size] = true;
-    else Reco_QQ_isCowboy[Reco_QQ_size] = false;
+	Reco_QQ_mupl_idx[Reco_QQ_size] = IndexOfThisMuon(&vMuon2 , Reco_mu_4mom, false);
+	Reco_QQ_mumi_idx[Reco_QQ_size] = IndexOfThisMuon(&vMuon1 , Reco_mu_4mom, false);
 
-    Reco_QQ_mupl_StationsMatched[Reco_QQ_size] = muon2->numberOfMatchedStations();
-    Reco_QQ_mumi_StationsMatched[Reco_QQ_size] = muon1->numberOfMatchedStations();
+	if (TVector2::Phi_mpi_pi(vMuon2.Phi() - vMuon1.Phi()) > 0) Reco_QQ_isCowboy[Reco_QQ_size] = true;
+	else Reco_QQ_isCowboy[Reco_QQ_size] = false;
 
-    Reco_QQ_mupl_isGoodMuon[Reco_QQ_size] = muon::isGoodMuon(*muon2, muon::TMOneStationTight);
-    Reco_QQ_mumi_isGoodMuon[Reco_QQ_size] = muon::isGoodMuon(*muon1, muon::TMOneStationTight);
+	iTrack_mupl = muon2->innerTrack();
+	iTrack_mumi = muon1->innerTrack();
```

Comment:
It is used to calculate the Reco_QQ_muXX_dXX_muonlessVtx variables, from a potential muonless Vertex refitting.
I will add the condition _muonLessPrimaryVertex on these lines, so that the iTrack_muXX variables are not set when it is not used.